mlflow/genai/optimize/optimizers/dspy_mipro_optimizer.py

```diff
@@ -186,8 +186,18 @@ def _maybe_suppress_stdout_stderr(self):
             yield
 
     def _display_optimization_result(self, program: "dspy.Predict"):
-        if hasattr(program, "score") and hasattr(program, "trial_logs"):
-            initial_score = program.trial_logs[1]["full_eval_score"]
+        score = getattr(program, "score", None)
+        if score is None:
+            return
+        # In DSPy < 2.6.17, trial_logs contains initial score in key=-1.
+        trial_logs = getattr(program, "trial_logs", {})
+        if 1 in trial_logs:
+            initial_score = trial_logs[1].get("full_eval_score")
+        elif -1 in trial_logs:
+            initial_score = trial_logs[-1].get("full_eval_score")
+        else:
+            initial_score = None
+        if initial_score is not None:
```

Comment:
```suggestion
        if initial_score is not None and initial_score != program.score:
```

do we need `initial_score != program.score`?